### PR TITLE
Move the NodeSet client into HD Daemon

### PR DESCRIPTION
**Pull request by jclapis**
This is part of the prep for Constellation and the nodeset.io v2 API. This reworks the NS client into a standalone library, and makes the HD daemon responsible for session management. The rest of the functions are proxied through the daemon so session management is done transparently.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -10,23 +10,27 @@ import (
...
 // Binder for the Hyperdrive daemon API server
 type ApiClient struct {
-	context client.IRequesterContext
-	Service *ServiceRequester
-	Tx      *TxRequester
-	Utils   *UtilsRequester
-	Wallet  *WalletRequester
+	context           client.IRequesterContext
+	NodeSet           *NodeSetRequester
+	NodeSet_StakeWise *NodeSetStakeWiseRequester
```
Comment (huy-nodeset) [inline client/client.go:15]: NodeSetStakeWise for consistency?

File: client/nodeset-stakewise.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// Gets the list of validators that the node has registered with the provided vault
+func (r *NodeSetStakeWiseRequester) GetRegisteredValidators(vault common.Address) (*types.ApiResponse[api.NodeSetStakeWise_GetRegisteredValidatorsData], error) {
+	args := map[string]string{
+		"vault": vault.Hex(),
+	}
+	return client.SendGetRequest[api.NodeSetStakeWise_GetRegisteredValidatorsData](r, "get-registered-validators", "GetRegisteredValidators", args)
+}
+
+// Gets the version of the latest deposit data set on the server for the provided vault
+func (r *NodeSetStakeWiseRequester) GetDepositDataSetVersion(vault common.Address) (*types.ApiResponse[api.NodeSetStakeWise_GetDepositDataSetData], error) {
```
Comment (huy-nodeset) [inline client/nodeset-stakewise.go:42]: Any reason why you're sharing this api response and not have its own GetDepositDataSetVersionData

**Review by huy-nodeset — APPROVED** [review]

File: client/nodeset-stakewise.go (hunk 2 above)
Comment (jclapis) [reply client/nodeset-stakewise.go]: Pure laziness, but I added one now.

File: client/client.go (hunk 1 above)
Comment (jclapis) [reply client/client.go:15]: I added an underscore mainly to distinguish that `_StakeWise` is actually a subrote of `NodeSet` (i.e. the route is `/api/v1/nodeset/stakewise`) - I'd expect to do that with any client requester that involves a subroute.

**Review by huy-nodeset — APPROVED** [review]

**Pull request merged**